pkg/minikube/command/exec_runner.go

```diff
@@ -85,12 +87,19 @@ func (*execRunner) RunCmd(cmd *exec.Cmd) (*RunResult, error) {
 }
 
 // Copy copies a file and its permissions
-func (*execRunner) Copy(f assets.CopyableFile) error {
+func (e *execRunner) Copy(f assets.CopyableFile) error {
 	dst := path.Join(f.GetTargetDir(), f.GetTargetName())
 	if _, err := os.Stat(dst); err == nil {
 		klog.Infof("found %s, removing ...", dst)
-		if err := os.Remove(dst); err != nil {
-			return errors.Wrapf(err, "error removing file %s", dst)
+		switch runtime.GOOS {
+		case "linux":
```

Comment:
AFAIK, execRunner should only run on Linux. What would you feel about escalating privilege only in cases of permission errors? For instance:

```
if err := os.Remove(dst); err != nil {
  if os.IsPermission(err) { 
    _, err = e.RunCmd(exec.Command("sudo", "rm", "-f", dst))
  }
  return errors.Wrapf(err, "error removing file %s", dst)
}

Reply:
Could this just call `.Remove()` to keep the logic in one place?

Reply:
Ok, i added the "isPermission" escalation and moved the code into the Remove() function